src/sophys/common/devices/pimega.py

```diff
@@ -146,6 +154,35 @@ class Pimega(SingleTrigger, PimegaDetector):
     def __init__(self, name, prefix, **kwargs):
         super(Pimega, self).__init__(prefix, name=name, **kwargs)
 
+    def stage(self):
+        # Make sure the current acquisition status is 'Done'
+        self._acquisition_signal.set(0).wait(timeout=30.0)
+
+        self._acquisition_signal.subscribe(
+            self._acquire_setpoint_changed, EpicsSignal.SUB_SETPOINT
+        )
+
+        return super().stage()
+
+    def unstage(self):
+        super().unstage()
+
+        self._acquisition_signal.unsubscribe(self._acquire_setpoint_changed)
+
+    def _acquire_setpoint_changed(self, value, severity, **kwargs):
+        if self._status is None or self._status.done:
+            return
+
+        if value == 1 and severity >= AlarmSeverity.MAJOR:
```

Comment:
Any specific reason for `MAJOR` instead of `MINOR`? Or `severity != 0`?

Reply:
From what I can see, asyn will usually set the severity to `INVALID`, *if* it isn't already set to something else. So it essentially considers any value as "indicating error".

Furthermore, while alarms are enumerated, their numerical value isn't necessarily increasing in severity (INVALID is simply different from MAJOR, it's not higher).

Either way, there won't be a case where the PV has worked correctly and it's set to `MINOR`. So to make the code simpler and not rely on "implementation details", I would simply use `severity != AlarmSeverity.NO_ALARM`

Reply:
Is that alarm behavior a Pimega-exclusive thing, an asyn-exclusive thing, or what? Checking [this documentation](https://docs.epics-controls.org/en/latest/process-database/EPICS_Process_Database_Concepts.html#alarm-severity), it seems indeed that MAJOR and INVALID should be treated as hard errors, while MINOR is more of a warning kind of thing.

Reply:
That would be mostly an asyn thing, for when the `write` function returns some sort of error.

That said, given the below quote, I think checking only for `INVALID` might be more correct (so using INVALID like this is not only an asyn thing, it's an alarm standard) :thinking: 
What do you think?

> For the other alarm types (i.e., scan, read/write), the severity is always INVALID and not configurable by the user.

`MINOR` and `MAJOR` alarms feel like they work better when you have analog readings and thresholds, where it makes sense to have a warning for something like a temperature increasing, and then a hazard when it reaches a critical value.